containers/src/Data/IntSet/Internal.hs

```diff
@@ -1060,41 +1060,69 @@ fromList xs
 -- | /O(n)/. Build a set from an ascending list of elements.
 -- /The precondition (input list is ascending) is not checked./
 fromAscList :: [Key] -> IntSet
-fromAscList [] = Nil
-fromAscList (x0 : xs0) = fromDistinctAscList (combineEq x0 xs0)
-  where
-    combineEq x' [] = [x']
-    combineEq x' (x:xs)
-      | x==x'     = combineEq x' xs
-      | otherwise = x' : combineEq x xs
+fromAscList = fromMonoList
```

Comment:
Should this be marked as `NOINLINE`?

This question also applies to `fromDistinctAscList` below, and to the `from*AscList*` functions in `Data.IntMap` and `Data.IntMap.Strict`.

Reply:
Probably.